PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -3253,17 +3272,99 @@ struct HfTrackIndexSkimCreatorLfCascades {
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1XiHyp);
+
+            auto secondaryVertex2XiHyp = df2.getPCACandidate();
+
+            if (std::abs(casc.mXi() - massXi) < cascadeMassWindow) {
+              SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi);
+            }
+
+            std::array<std::array<float, 3>, 2> arrMomToXi = {pVecXi, pVecPion1XiHyp};
+            auto mass2ProngXiHyp = RecoDecay::m(arrMomToXi, arrMass2Prong[hf_cand_casc_lf::DecayType2Prong::XiczeroOmegaczeroToXiPi]);
+
+            if (mass2ProngXiHyp < massXiPiMin || mass2ProngXiHyp > massXiPiMax) {
+              statusMassXiPiChannel = false;
+            }
+
+            // fill histograms
+            if (fillHistograms && statusMassXiPiChannel) {
+              registry.fill(HIST("hVtx2ProngXXiHyp"), secondaryVertex2XiHyp[0]);
+              registry.fill(HIST("hVtx2ProngYXiHyp"), secondaryVertex2XiHyp[1]);
+              registry.fill(HIST("hVtx2ProngZXiHyp"), secondaryVertex2XiHyp[2]);
+              registry.fill(HIST("hMassXicZeroOmegacZeroToXiPi"), mass2ProngXiHyp);
+            }
+          }
+
+          // find charm baryon decay using omega PID hypothesis
+          int nVtxFrom2ProngFitterOmegaHyp = df2.process(trackCascOmega, trackParVarPion1);
+          if (nVtxFrom2ProngFitterOmegaHyp > 0) {
+
+            statusMassOmegaPiChannel = true;
+
+            df2.propagateTracksToVertex();
+
+            std::array<float, 3> pVecOmega = {0.};
+            std::array<float, 3> pVecPion1OmegaHyp = {0.};
+            df2.getTrack(0).getPxPyPzGlo(pVecOmega);
+            df2.getTrack(1).getPxPyPzGlo(pVecPion1OmegaHyp);
+
+            auto secondaryVertex2OmegaHyp = df2.getPCACandidate();
+
+            if (std::abs(casc.mOmega() - massOmega) < cascadeMassWindow) {
+              SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi);
+            }
+
+            std::array<std::array<float, 3>, 2> arrMomToOmega = {pVecOmega, pVecPion1OmegaHyp};
+            auto mass2ProngOmegaHyp = RecoDecay::m(arrMomToOmega, arrMass2Prong[hf_cand_casc_lf::DecayType2Prong::OmegaczeroToOmegaPi]);
+
+            if (mass2ProngOmegaHyp < massOmegaPiMin || mass2ProngOmegaHyp > massOmegaPiMax) {
+              statusMassOmegaPiChannel = false;
+            }
+
+            // fill histograms
+            if (fillHistograms && statusMassOmegaPiChannel) {
+              registry.fill(HIST("hVtx2ProngXOmegaHyp"), secondaryVertex2OmegaHyp[0]);
+              registry.fill(HIST("hVtx2ProngYOmegaHyp"), secondaryVertex2OmegaHyp[1]);
+              registry.fill(HIST("hVtx2ProngZOmegaHyp"), secondaryVertex2OmegaHyp[2]);
+              registry.fill(HIST("hMassOmegacZeroToOmegaPi"), mass2ProngOmegaHyp);
+            }
+          }
+
+          // fill table row only if a vertex was found
+          if ((nVtxFrom2ProngFitterXiHyp != 0 && statusMassXiPiChannel) || (nVtxFrom2ProngFitterOmegaHyp != 0 && statusMassOmegaPiChannel)) {
```

Comment:
No need to test `nVtxFrom2ProngFitterXiHyp != 0` because the flag is also `0` if there is no vertex found. Moreover, `statusMassXiPiChannel` is somehow redundant (you can simply check if `hfFlag != 0`